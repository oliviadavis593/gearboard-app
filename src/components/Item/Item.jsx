import React, { Component } from 'react';
import GearContext from '../../GearContext';
import EditItem from '../EditItem/EditItem';


class Item extends Component {

    static contextType = GearContext; 

    render() {
        const { rating, gearName, description, comment } = this.props; 
        //console.log(this.context.deleteItem)
        return(
            <div className='item'>
                <h2 className='item__name'>
                    {gearName}
                </h2>
                <div className='item__rating'>
                    Rating: {rating}
                </div>
                <div className='item__description'>
                   description: {description}
                </div>
                <div className='item__comment'>
                   comment: {comment}
                </div>
                <EditItem 
                edit='Edit Item'
                />
                <div className='gb-list-delete__container'>
                    <button
                    className='gb-item__delete'
                    type='button'
                    onClick={() => this.context.deleteItem(this.props.id)}
                    >
                        Delete
                    </button>
                </div>
            </div>
        )
    }
}

export default Item; 