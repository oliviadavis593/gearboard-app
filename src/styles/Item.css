.gb-item {
    --item-height: 88px;
    background-color: #eee;
    padding: 8px 0px 8px 8px; 
    margin: 0 20px 15px -20px;
}

.gb-item h2{ 
    font-size: 25px; 
}

.gb-item a{
    text-decoration: none; 
    color: #000; 
}

.gb-item:hover a{
    color: #e67e22; 
    transition: 0.9s; 
}

