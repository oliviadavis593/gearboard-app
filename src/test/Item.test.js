import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Item from '../components/Item/Item';


describe.only('Item component', () => {
    it(`renders without crashing`, () => {
        const div = document.createElement('div')
        ReactDOM.render(<Item gear_name='whatever' id={2}/>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<Item gear_name='test gear_name' id={1}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})