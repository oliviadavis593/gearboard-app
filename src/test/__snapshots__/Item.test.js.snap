// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component renders the UI as expected 1`] = `
<div
  className="gb-item"
>
  <a
    href="/item/undefined"
    onClick={[Function]}
  >
    <h2
      className="gb-item__gear_name"
    >
      test gear_name
    </h2>
  </a>
  <div
    className="gb-list-delete__container"
  >
    <button
      className="gb-item__delete"
      onClick={[Function]}
      type="button"
    >
       
      Delete
    </button>
  </div>
</div>
`;
